resources/dw/virtualwarehouse/impala/resource_impala_vw_acc_test.go

```diff
@@ -0,0 +1,152 @@
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/cdpacctest"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type impalaTestParameters struct {
+	Name              string
+	ClusterID         string
+	DatabaseCatalogID string
+	ImageVersion      string
+}
+
+func ImpalaPreCheck(t *testing.T) {
+	errMsg := "AWS CDW Impala Terraform acceptance testing requires environment variable %s to be set"
+	if _, ok := os.LookupEnv("CDW_CLUSTER_ID"); !ok {
+		t.Skipf(errMsg, "CDW_CLUSTER_ID")
+	}
+	if _, ok := os.LookupEnv("CDW_DATABASE_CATALOG_ID"); !ok {
+		t.Skipf(errMsg, "CDW_DATABASE_CATALOG_ID")
+	}
+}
+
+func TestAccImpala_basic(t *testing.T) {
+	params := impalaTestParameters{
+		Name:              cdpacctest.RandomShortWithPrefix(cdpacctest.ResourcePrefix),
+		ClusterID:         os.Getenv("CDW_CLUSTER_ID"),
+		DatabaseCatalogID: os.Getenv("CDW_DATABASE_CATALOG_ID"),
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			cdpacctest.PreCheck(t)
+			ImpalaPreCheck(t)
+		},
+		ProtoV6ProviderFactories: cdpacctest.TestAccProtoV6ProviderFactories,
+		CheckDestroy:             testCheckImpalaDestroy,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: utils.Concat(
+					cdpacctest.TestAccCdpProviderConfig(),
+					testAccImpalaBasicConfig(params)),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "name", params.Name),
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "cluster_id", params.ClusterID),
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "database_catalog_id", params.DatabaseCatalogID),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccImpalaBasicConfig(params impalaTestParameters) string {
+	return fmt.Sprintf(`
+		resource "cdp_dw_vw_impala" "test_impala" {
+		  cluster_id = %[1]q
+		  database_catalog_id = %[2]q
+		  name = %[3]q
+		  image_version = %[4]q
+		}
+	`, params.ClusterID, params.DatabaseCatalogID, params.Name, params.ImageVersion)
+}
+
+func testCheckImpalaDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cdp_dw_vw_impala" {
+			continue
+		}
+
+		cdpClient := cdpacctest.GetCdpClientForAccTest()
+		params := operations.NewDescribeVwParamsWithContext(context.Background())
+		clusterID := rs.Primary.Attributes["cluster_id"]
+		params.WithInput(&models.DescribeVwRequest{
+			VwID:      &rs.Primary.ID,
+			ClusterID: &clusterID,
+		})
+
+		_, err := cdpClient.Dw.Operations.DescribeVw(params)
+		if err != nil {
+			if strings.Contains(err.Error(), "Virtual Warehouse not found") {
+				continue
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+// Set RUN_IMAGE_VERSION_TESTS to run this test
```

Comment:
why? :) this test runs already with a build flag condition, why do you need another flag? It is not a bad thing to test the versions

Reply:
When this test runs, you have to explicitly enter the image version as the hardcoded image version would fail. As such wanted to make it explicit and different from basic so had to do that but if you feel I should remove it, that also sounds good.

Reply:
Okay, added the code to fetch latest image version and use it